### PR TITLE
[INLONG-3800][Sort] Add GroupInfo and StreamInfo definition to support transform

### LINKED ISSUE #3800: [Feature][Sort] Add GroupInfo,StreamInfo definition to support transform
### Description

Add GroupInfo,StreamInfo definition to support transform

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Title Name: [INLONG-3800][Sort] Add GroupInfo,StreamInfo definition to support transform

Fixes #3800

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/GroupInfo.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.io.Serializable;
+import java.util.List;
+
+@Data
+public class GroupInfo implements Serializable {
+
+    private static final long serialVersionUID = 6034630524669634079L;
+
+    @JsonProperty("groupId")
+    private String id;
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/GroupInfo.java:34]: Why not name this variable as `groupId`?

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/StreamInfo.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.Serializable;
+import java.util.List;
+
+@Data
+public class StreamInfo implements Serializable {
+
+    private static final long serialVersionUID = 82342770067926123L;
+
+    @JsonProperty("streamId")
+    private String id;
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/StreamInfo.java:36]: Why not name this variable as `streamId`?

File: inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/GroupInfoTest.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+import org.apache.inlong.sort.protocol.transformation.relation.NodeRelationShip;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import static org.junit.Assert.assertEquals;
+
+public class GroupInfoTest {
```
Comment (healchow) [inline inlong-sort/sort-common/src/test/java/org/apache/inlong/sort/protocol/GroupInfoTest.java:44]: Please add some java docs for the class and public methods.

**Review by healchow — COMMENTED** [review]
PTAL.

**Review by gong — APPROVED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/GroupInfo.java (hunk 1 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/GroupInfo.java:34]: The id is defined for internal use. The serialized block has been defined as groupId. Of course, it is best to name this block uniformly.

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/StreamInfo.java (hunk 2 above)
Comment (yunqingmoswu) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/StreamInfo.java:36]: The id is defined for internal use. The serialized block has been defined as streamId. Of course, it is best to name this block uniformly.

**Review by EMsnap — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Pull request merged**